app/product/[handle]/page.tsx

```diff
@@ -28,8 +28,8 @@ export async function generateMetadata({
   const hide = !product.tags.includes(HIDDEN_PRODUCT_TAG);
 
   return {
-    title: product.seo.title || product.title,
-    description: product.seo.description || product.description,
+    title: product.seo?.title || product.title,
+    description: product.seo?.description || product.description,
```

Comment:
todo: We don't have a `seo` prop, so we can remove that
```suggestion
    title: product.title,
    description: product.description,
```